c10/core/StorageImpl.h

```diff
@@ -0,0 +1,227 @@
+#pragma once
+
+#include <c10/core/Allocator.h>
+#include <c10/core/ScalarType.h>
+#include <c10/core/ScalarTypeUtils.h>
+
+#include <c10/util/intrusive_ptr.h>
+
+namespace at {
+struct Type;
```

Comment:
You should have gotten rid of the Type forward declaration *before* this patch ;) Now you get to hear me complain about it.